other/no-loss-lottery/lottery/src/lib.rs

```diff
@@ -0,0 +1,321 @@
+    "functions": [
+      {
+        "name": "new",
+        "inputs": [
+          {
+            "type": "Custom",
+            "name": "scrypto::resource::Bucket",
+            "generics": []
+          }
+        ],
+        "output": {
+          "type": "Custom",
+          "name": "scrypto::core::Component",
+          "generics": []
+        }
+      }
+    ],
+    "methods": [
+      {
+        "name": "new_user",
+        "mutability": "Immutable",
+        "inputs": [],
+        "output": {
+          "type": "Custom",
+          "name": "scrypto::resource::Bucket",
+          "generics": []
+        }
+      },
+      {
+        "name": "stake",
+        "mutability": "Mutable",
+        "inputs": [
+          {
+            "type": "Custom",
+            "name": "scrypto::resource::BucketRef",
+            "generics": []
+          },
+          {
+            "type": "Custom",
+            "name": "scrypto::resource::Bucket",
+            "generics": []
+          }
+        ],
+        "output": {
+          "type": "Unit"
+        }
+      },
+      {
+        "name": "account_reward",
+        "mutability": "Mutable",
+        "inputs": [
+          {
+            "type": "Custom",
+            "name": "scrypto::resource::BucketRef",
+            "generics": []
+          }
+        ],
+        "output": {
+          "type": "Custom",
+          "name": "scrypto::types::Decimal",
+          "generics": []
+        }
+      },
+      {
+        "name": "withdraw",
+        "mutability": "Mutable",
+        "inputs": [
+          {
+            "type": "Custom",
+            "name": "scrypto::resource::BucketRef",
+            "generics": []
+          }
+        ],
+        "output": {
+          "type": "Custom",
+          "name": "scrypto::resource::Bucket",
+          "generics": []
+        }
+      },
+      {
+        "name": "get_reward",
+        "mutability": "Mutable",
+        "inputs": [
+          {
+            "type": "Custom",
+            "name": "scrypto::resource::BucketRef",
+            "generics": []
+          }
+        ],
+        "output": {
+          "type": "Custom",
+          "name": "scrypto::resource::Bucket",
+          "generics": []
+        }
+      }
+    ]
+  }
+"# }
+
+
+#[derive(NftData)]
+pub struct LotteryCard {
+    lottery_name: String,
+    lottery_epoch: u64
+}
+
+blueprint! {
+    struct Lottery {
+        // staking component
+        staking: Staking,
+        // admin and minting badges
+        admin_badge: ResourceDef,
+        // vault for minting tickets
+        ticket_minter: Vault,
+        // address of staking contract
+        staking_address: Address,
+        // staking user badge
+        staking_user: Vault,
+        // vault that will be used for withdrawals
+        withdraw_vault: Vault,
+        // resource def for future minting
+        tickets: ResourceDef,
+        // players staking amount
+        players: HashMap<Address, HashMap<String, Decimal>>,
```

Comment:
Having maps with account addresses as key is an anti-pattern we are trying to avoid. You could instead store the `HashMap<String, Decimal>` inside the player's ticket metadata